### ISSUE TITLE
redundant_pattern_matching does not check for match with single arm

### ISSUE THREAD

**Issue by disco07** [rust-lang/rust-clippy#10726]
### Summary

Use of match, matching `Result` or `Option`;  do not trigger `clippy::redundant_pattern_matching`

### Reproducer

I tried this code:

```rust
match Err::<i32, i32>(55) {
        Ok(_) => true,
        _ => false,
    };
```

I expected to see this happen:
```rust
Err::<i32, i32>(55).is_ok();
```

Instead, this happened:
```rust
match Err::<i32, i32>(55) {
        Ok(_) => true,
        _ => false,
    };
```

### Version

```text
rustc 1.71.0-nightly (b628260df 2023-04-22)
binary: rustc
commit-hash: b628260df0587ae559253d8640ecb8738d3de613
commit-date: 2023-04-22
host: x86_64-unknown-linux-gnu
release: 1.71.0-nightly
LLVM version: 16.0.2
```

**Comment by disco07** [comment]
@xFrednet

**Comment by xFrednet** [comment]
Hey @disco07, thank you for the ticket :). Do you have a specific question related to it :)

**Comment by disco07** [comment]
**@xFrednet** Can I work on this ?

**Comment by xFrednet** [comment]
Yes, you're very welcome to! As a start, you can take a look at the lint code: https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/matches/redundant_pattern_match.rs

And look at the AST structure of the code you want to detect. For this, I recommend using `#[clippy::dump]` and running `tool > clippy`. See [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4280d721d559f645ef063ca6f3a92924). Example actually triggers the [`match_like_matches_macro`](https://rust-lang.github.io/rust-clippy/master/index.html#match_like_matches_macro) lint.

I guess that the lint ignores the match, as it expects the `Ok` and `Err` branches, and not a wildcard. Maybe you just need to add a check for that :)
